enterprise/dev/deployment-notifications/main.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/google/go-github/v41/github"
+	"github.com/slack-go/slack"
+	"github.com/sourcegraph/sourcegraph/dev/sg/root"
+	"github.com/sourcegraph/sourcegraph/dev/team"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"golang.org/x/oauth2"
+)
+
+type Flags struct {
+	SourcegraphCommit      string
+	MockLiveCommit         string
+	GitHubToken            string
+	Pretend                bool
+	GuessSourcegraphCommit bool
+	Environment            string
+	SlackToken             string
+	SlackAnnounceWebhook   string
+}
+
+func (f *Flags) Parse() {
+	flag.StringVar(&f.GitHubToken, "github.token", os.Getenv("GITHUB_TOKEN"), "mandatory github token")
+	flag.StringVar(&f.SourcegraphCommit, "sourcegraph.commit", "", "Sourcegraph commit being deployed")
+	flag.StringVar(&f.Environment, "environment", "", "Environment being deployed")
+	flag.StringVar(&f.MockLiveCommit, "mock.live-commit", "", "Use this commit instead of requesting the commit deployed on the target environment")
+	flag.BoolVar(&f.Pretend, "pretend", false, "Pretend to post notifications, printing to stdout instead")
+	flag.BoolVar(&f.GuessSourcegraphCommit, "sourcegraph.guess-commit", false, "Attempt at inferring the deployed commit from the changes in the diff")
```

Comment:
```suggestion
	flag.BoolVar(&f.InferSourcegraphCommit, "sourcegraph.infer-commit", false, "Attempt at inferring the deployed commit from the changes in the diff")
```

You already use `infer` here and I think it's the much better term!